formatter/formatter.go

```diff
@@ -136,6 +148,48 @@ func JUnitReportXML(report *parser.Report, noXMLHeader bool, goVersion string, w
 	return nil
 }
 
+func mergeBenchmarks(benchmarks []*parser.Benchmark) []*parser.Benchmark {
```

Comment:
The total number of benchmarks is known, so we can just divide by that rather than having to keep track of a moving average. Also, by reusing the `Benchmark` type we won't have to use a map of maps. What about something like this:

```go
func mergeBenchmarks(benchmarks []*parser.Benchmark) []*parser.Benchmark {                                             
    var merged []*parser.Benchmark                                                                                     
    benchmap := make(map[string][]*parser.Benchmark)                                                                   
    for _, bm := range benchmarks {                                                                                    
        if _, ok := benchmap[bm.Name]; !ok {                                                                           
            merged = append(merged, &parser.Benchmark{Name: bm.Name})                                                  
        }                                                                                                              
        benchmap[bm.Name] = append(benchmap[bm.Name], bm)                                                              
    }                                                                                                                  
                                                                                                                       
    for _, bm := range merged {                                                                                        
        for _, b := range benchmap[bm.Name] {                                                                          
            bm.Allocs += b.Allocs                                                                                      
            bm.Bytes += b.Bytes                                                                                        
            bm.Duration += b.Duration                                                                                  
        }                                                                                                              
        n := len(benchmap[bm.Name])                                                                                    
        bm.Allocs /= n                                                                                                 
        bm.Bytes /= n                                                                                                  
        bm.Duration /= time.Duration(n)                                                                                
    }                                                                                                                  
    return merged                                                                                                      
}   
```